src/single-spa-react.js

```diff
@@ -207,30 +191,19 @@ function atLeastReact16(React) {
 function reactDomRender({ opts, elementToRender, domElement }) {
   const renderType =
     typeof opts.renderType === "function" ? opts.renderType() : opts.renderType;
-  if (
-    [
-      "createRoot",
-      "unstable_createRoot",
-      "createBlockingRoot",
-      "unstable_createBlockingRoot",
-    ].indexOf(renderType) >= 0
-  ) {
-    const root = opts.ReactDOM[renderType](domElement);
-    root.render(elementToRender);
+
+  if (renderType === "hydrateRoot") {
+    const root = opts.ReactDOMClient[renderType](domElement, elementToRender);
```

Comment:
The migration docs made this specific callout:

> // Unlike with createRoot, you don't need a separate root.render() call here.

See also: https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-client-rendering-apis